### ISSUE TITLE
Catch and Save Syntax Error message

### ISSUE THREAD

**Issue by antigens** [josdejong/mathjs#1227]
-- v5.1.1 Latest

So I have this calculator widget that I'm working on and I'm trying to catch the error messages which I will display in an HTML element upon or after the error occurs. Here is the structure of my code with comments on what I'm looking to achieve:

```
dataController {
    data: {
         error: undefined,
         // more data
    }
 }

UIController {
    // handle UI events
 }

parseController {
    // parse functions
    return: method () {
        // validate input via math.parse()
    }
 }

appController {

    getResult() { 
        // parseController.method(input) /* Here is where the code stops after Syntax Error */

        /** catch error > save to error data controller

              Ideally, I'd like to catch the error somewhere here and pass it to the
              dataController so I can handle the error display function for the UI

         **/
       
    }

 }

app.init;
```

Is there any way that I can catch the error message? I've read a few SO issues ([relevant](https://stackoverflow.com/questions/5963045/can-syntax-errors-be-caught-in-javascript)) which state that we can't catch SyntaxError's since they are thrown during code parsing and not during runtime, therefore, `try...catch` wouldn't work.

I tried it initially, but as soon as the `parse()` or `eval()` methods are hit on the try block, the error is thrown immediately in the console without catching it.

I looked into the source code but it seems like the `createSyntaxError()` function cannot be extended as it is private and is called on the fly by different parsing functions.

Can you suggest a different or a better way of catching error messages?

**Comment by josdejong** [comment]
mathjs has it's own expression parser so you can indeed catch syntax errors simply using `try...catch`. 

```
try { 
  math.eval('2 / / 3')
} catch (err) {
  console.log(err)
}
```
will log a syntax error `SyntaxError: Value expected (char 5)`

**Comment by antigens** [comment]
Ok, `try...catch` indeed works this time, now I feel dumb.. guess I messed my catch solution is why it wouldn't work before.

Also found another solution on capturing the error message by `window.onerror` event. I found this working while I still can't figure out the former issue.

Something like:
```
window.addEventListener('error', function(err) {
    data.error = err.message // ErrorEvent Message
});
```

Though I'd probably just use `try...catch` since it's a bit cleaner.

Closing this issue now. Thanks man!
